src/core/constructs/components.py

```diff
@@ -10,7 +10,7 @@
 
 from pydantic import BaseModel
 
-from .resource import ResourceReferenceModel, ResourceModel
+from .resource import ResourceReferenceModel, TaggableResourceModel
```

Comment:
The component construct should still work with the `ResourceModel` class. Since `TaggableResourceModel` is a subclass it won't make a difference, and this way the component works on the highest level class as possible.